src/main/java/org/dice_research/opal/civet/metrics/AccessibilityMetric.java

```diff
@@ -0,0 +1,102 @@
+package org.dice_research.opal.civet.metrics;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.Socket;
+import java.net.URL;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.NodeIterator;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.ResourceFactory;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.vocabulary.DCAT;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dice_research.opal.civet.Metric;
+import org.dice_research.opal.common.vocabulary.Opal;
+
+/**
+ * The Accessibility awards stars based on the Accessibility of metadata.
+ * A connection is made using the URL of metadata and ratings are evaluated based on returned status code.
+ * 
+ * @author Amit Kumar
+ */
+//HTTP 200: “Everything is OK.”
+
+//401: “Unauthorized” or “Authorization Required.”
+//403: “Access to that resource is forbidden.”
+//404: “The requested resource was not found.” 
+//499: “Client closed request.”
+
+public class AccessibilityMetric implements Metric {
+
+	private static final Logger LOGGER = LogManager.getLogger();
+	private static final String DESCRIPTION = "Computes the quality of dataset as per contactability metric." 
+			+ "Four kinds of ratings are awarded to the dataset which are following: "
+			+ "5 Stars: If the URL is in the dataset and returns a successfull code HTTP 200: “Everything is OK.” upon making connection."
+			+ "2 Stars: If the URL is in the dataset and returns a status code HTTP 401: “Unauthorized” or “Authorization Required.” "
+			+ "1 Stars: If the URL is in the dataset and it return a status code 403: “Access to that resource is forbidden.” "
+			+ "0 Stars: If the URL is not found in the dataset.";
+
+
+
+	@Override
+	public Integer compute(Model model, String datasetUri) throws Exception {
+		LOGGER.info("Processing dataset " + datasetUri);
+		Resource distribution = ResourceFactory.createResource(datasetUri);
+		Statement statement = model.getProperty(distribution, DCAT.accessURL);
```

Comment:
I have a small doubt. The datasetUri is an URI for a dataset. 

Resource distribution = ResourceFactory.createResource(datasetUri); ---> is a Dataset Resource.

Statement statement = model.getProperty(distribution, DCAT.accessURL); ---> distribution here is actually a dataset and AccessURL comes under the domain dcat:Distribution. 

We will never get any statement at all with real world data.

A solution would be that we check for distributions in the provided dataset and then check if those distributions have AccessURL as a property.